x/crosschain/keeper/msg_server_migrate_erc20_custody_funds.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// MigrateERC20CustodyFunds migrates the funds from the current TSS to the new TSS
+func (k msgServer) MigrateERC20CustodyFunds(
+	goCtx context.Context,
+	msg *types.MsgMigrateERC20CustodyFunds,
+) (*types.MsgMigrateERC20CustodyFundsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check if authorized
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, errorsmod.Wrap(authoritytypes.ErrUnauthorized, err.Error())
```

Comment:
Shouldn't this be
```suggestion
		return nil, errorsmod.Wrap(err, "unable to check authorization")
```
?

Reply:
Yeah this is something we have throughout the codebase and we need to improve, I would keep it as it in this PR for consistency with other files but created issue for this matter: https://github.com/zeta-chain/node/issues/2678